server/handlers/clients.js

```diff
@@ -165,28 +187,37 @@ function clientsRouter(clients, defaultId) {
     }
 
     res.status(200).json(info);
-  });
+  };
+}
 
-  router.post('/:id', async (req, res) => {
+function addConfigHandler(logger, bpanelConfig) {
+  if (!configsMap) {
+    const clientConfigs = loadClientConfigs(bpanelConfig);
+    configsMap = createConfigsMap(clientConfigs);
+  }
+  return (req, res) => {
     const id = req.params.id;
-    if (clientInfo[id])
+    if (configsMap.get(id))
       return res
         .status(409)
         .send({ message: `A client with the id '${id}' already exists` });
 
-    return updateOrAdd(req, res);
-  });
+    return updateOrAdd(logger, req, res);
+  };
+}
 
-  router.delete('/:id', (req, res) => {
+function updateConfigHandler(logger) {
+  return (req, res) => updateOrAdd(logger, req, res);
```

Comment:
I think that the first middleware should attach the logger to the request

Reply:
so just `req.logger = logger`? I suppose we could do that for `bpanelConfig` too then. I was considering this but thought it might feel a little dirty to pollute the request object with non-request related things, but it would definitely simplify and streamline the code.